packages/alfa-table/src/microsyntaxes.ts

```diff
@@ -0,0 +1,36 @@
+import { Err, Ok, Result } from "@siteimprove/alfa-result";
+
+/**
+ * @see https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#signed-integers
+ */
+export function parseInteger(str: string): Result<number, string> {
+  // empty/whitespace string are errors for specs, not for Number…
+  // \s seems to be close enough to "ASCII whitespace".
+  if (str.match(/^\s*$/)) {
+    return Outcomes.empty;
+  }
+  const raw = Number(str);
+  return isNaN(raw)
+    ? Outcomes.notNumber
+    : raw !== Math.floor(raw)
+    ? Outcomes.notInteger
+    : // 0 and -0 are different (but equal…) floats. Normalising.
+      Ok.of(raw === -0 ? 0 : raw);
```

Comment:
This needs to properly parse an integer according to the HTML specification rather than rely on the rather odd number conversion rules of JavaScript.